standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -708,6 +734,13 @@ private void removeTxnComponents(Connection dbConn, CompactionInfo info) throws
     PreparedStatement pStmt = null;
     ResultSet rs = null;
     try {
+      /*
+       * Remove all retry associated metadata of table/partition in the TXN_CLEANUP_QUEUE both when compaction
```

Comment:
isn't it specific only to aborts? how is this related to regular compaction requests?

Reply:
@deniskuzZ Compaction can clean abort files, hence removing associated TXN_COMPONENTS records. This means we can have a situation wherein there are no TXN_COMPONENTS records but we could have retry records. To ensure this, we need to remove the retry records if present whenever compaction on the table/partition is successful.

Reply:
got it. still it's kinda unexpected to find that inside txnComponents cleanup we do some extra stuff